### PR TITLE
Update the analyze databases article

### PR DESCRIPTION
**Pull request by aeisenberg**
This change updates the analyze databases article to clarify examples. It reorganizes to put packs examples first and rearranges a few paragraphs.

### CODE REVIEW

File: docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst [hunk 1]
```diff
@@ -97,53 +101,92 @@ the `database analyze reference documentation <../manual/database-analyze>`__.
...
 
-The following examples assume your CodeQL databases have been created in a
-directory that is a sibling of your local copies of the CodeQL and CodeQL for Go
-repositories.
+Specifying which queries to run in a CodeQL pack
+------------------------------------------------
 
-Running a single query
-~~~~~~~~~~~~~~~~~~~~~~
+*Query specifiers* are used by ``codeql database analyze`` and other commands that operate on a set of queries.
```
Comment (jf205) [inline docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst:109]:
```suggestion
Query specifiers are used by ``codeql database analyze`` and other commands that operate on a set of queries.
```

File: docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst [hunk 2]
```diff
@@ -34,7 +38,7 @@ When you run ``database analyze``, it:
 
 You can analyze a database by running the following command::
 
-   codeql database analyze <database> --format=<format> --output=<output> <queries>
+   codeql database analyze <database> --format=<format> --output=<output> ...<query-specifiers>
```
[the remark this answers is not in the document]
Comment (aeisenberg) [reply docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst:41]: The `...` indicates that there can be multiple query specifiers. There should be some visual indicator that multiple specifiers can be used, but maybe `...` isn't best.

File: docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst [hunk 3]
```diff
@@ -52,7 +56,7 @@ You must specify:
 
 You can also specify:
 
-- ``...<query-specifications>``: a list of queries to run over your database. This
+- ``...<query-specifiers>``: a list of queries to run over your database. This
```
[the remark this answers is not in the document]
Comment (aeisenberg) [reply docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst:59]: I'll follow what we do for database analyze. I also want to add a few words.

```suggestion
- ``<query-specifiers>...``: a spece-separated list of queries to run over your database. This
```

File: docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (aeisenberg) [reply docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst:41]: The docs for [database analyze](https://codeql.github.com/docs/codeql-cli/manual/database-analyze/) uses `...` _after_ the parameter. I'll do the same.

```suggestion
   codeql database analyze <database> --format=<format> --output=<output> <query-specifiers>...
```

**Review by aeisenberg — COMMENTED** [review]
Thanks for the review!

File: docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (jf205) [reply docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst]: 👍🏻 

Note there's a typo: spece-separated -> space-separated

[the remark this answers is not in the document]
Comment (aeisenberg) [reply docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst]: 😢

File: docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst [hunk 4]
```diff
@@ -52,8 +56,8 @@ You must specify:
 
 You can also specify:
 
-- ``...<query-specifications>``: a list of queries to run over your database. This
-  is a list of arguments. Where each argument can be:
+- ``<query-specifiers>...``: a spece-separated list of queries to run over your database. This
```
Comment (aeisenberg) [inline docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst:59]:
```suggestion
- ``<query-specifiers>...``: a space-separated list of queries to run over your database. This
```

**Comment by aeisenberg** [comment]
Got an [odd failure](https://github.com/github/semmle-code/actions/runs/3099334282/jobs/5018395605#step:13:587) when running the docs job

> HTTP request sent, awaiting response... 503 Egress is over the account limit.
> 2022-09-21 16:21:56 ERROR 503: Egress is over the account limit..

Rerunning.

**Comment by aeisenberg** [comment]
Thanks for the fix @jf205. Is this good to merge now?

**Comment by jf205** [comment]
LGTM, but let's get a review from the docs team too. I've added the magic label.

**Comment by aeisenberg** [comment]
@github/docs-content is there anyone able to take a look at this? Thank you.

File: docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst [hunk 5]
```diff
@@ -15,11 +15,15 @@ For information about writing queries to run with ``database analyze``, see
 
 Before starting an analysis you must:
 
-- :doc:`Set up the CodeQL CLI <getting-started-with-the-codeql-cli>` so that it can find the queries
-  and libraries included in the CodeQL repository.
+- :doc:`Set up the CodeQL CLI <getting-started-with-the-codeql-cli>` to run commands locally and
+  optionally check out the CodeQL repository if you want direct access to the CodeQL core queries
```
Comment (adityasharad) [inline docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst:19]: Avoid highlighting the idea of checking out the repo here. If they are using a bundle or packs there is no need.

File: docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst [hunk 6]
```diff
@@ -95,55 +99,93 @@ You can also specify:
...
+pack.
 
-You can also run your own custom queries with the ``database analyze`` command.
-For more information about preparing your queries to use with the CodeQL CLI,
-see ":doc:`Using custom queries with the CodeQL CLI <using-custom-queries-with-the-codeql-cli>`."
+Example query specifiers
+~~~~~~~~~~~~~~~~~~~~~~~~
 
-If you do not have the CodeQL repository checked out, you can execute the same queries by specifying the query pack name and the path to the queries::
+* ``codeql/python-queries`` - All the queries in the default query suite of the latest version of the ``codeql/python-queries`` pack.
```
Comment (adityasharad) [inline docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst:133]: Should we mention that the default suite is `python-code-scanning.qls`?

File: docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst [hunk 7]
```diff
@@ -224,33 +270,28 @@ For more information about CodeQL packs, see :doc:`About CodeQL Packs <about-cod
...
-the CodeQL repository::
+For more information, see ":ref:`Creating CodeQL query suites <creating-codeql-query-suites>`."
 
-   ql/<language>/ql/src/codeql-suites/<language>-code-scanning.qls
+Query suites in the core CodeQL query packs are all stored in the ``codeql-suites`` directory of the pack. For example, the ``codeql/cpp-queries`` pack contains the following query suites:
 
-and at the following path in the CodeQL for Go repository::
+* ``cpp-code-scanning.qls`` - Standard Code Scanning queries for C++.
+* ``cpp-security-and-quality`` - Security-and-quality queries for C++.
+* ``cpp-security-extended`` - Security-extended queries for C++. This suite contains queries that are less precise than the standard security queries, and may find more false-positives.
```
Comment (adityasharad) [inline docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst:292]: Reorder these: security-extended should go first, and highlight that each includes the suites above it.

**Review by adityasharad — COMMENTED** [review]
Looks good, some minor suggestions.

File: docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst (hunk 5 above)
Comment (aeisenberg) [reply docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst:19]: Good point.

File: docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst [hunk 8]
```diff
@@ -95,55 +97,105 @@ You can also specify:
...
+.. pull-quote::
+
+    Tip
+
+    The default query suite of the standard CodeQL query packs are ``codeql-suites/<lang>-code-scanning.qls``. Several other useful query suites can also be found in the ``codeql-suites`` directory of each pack. For example, the ``codeql/cpp-queries`` pack contains the following query suites:
 
-Use the ``--download`` flag to download the query pack if it isn't yet available locally.
+    * ``cpp-code-scanning.qls`` - Standard Code Scanning queries for C++. The default query suite for this pack.
+    * ``cpp-security-extended.qls`` - Security-extended queries for C++. This suite contains queries that are less precise than the standard security queries, and may find more false-positives. This query suite includes all queries from ``cpp-code-scanning.qls``.
+    * ``cpp-security-and-quality.qls`` - Security-and-quality queries for C++. This query suite includes all queries from ``cpp-security-extended.qls``.
```
Comment (adityasharad) [inline docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst:147]: Adapted the descriptions from https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning#running-additional-queries.
```suggestion
    * ``cpp-security-extended.qls`` - Queries from the default  ``cpp-code-scanning.qls`` suite for C++, plus lower severity and precision queries.
    * ``cpp-security-and-quality.qls`` - Queries from ``cpp-security-extended.qls``, plus maintainability and reliability queries.
```

**Review by adityasharad — APPROVED** [review]
Good stuff. I'm happy with the content, over to docs team for final editorial review.

File: docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst [hunk 9]
```diff
@@ -95,55 +97,105 @@ You can also specify:
...
+containing one or more queries, or a ``.qls`` query suite file. If
+you omit a pack name, then you must provide a ``path``,
+which will be interpreted relative to the working directory
+of the current process. Glob patterns are not supported.
 
-The analysis generates a CSV file (``js-results.csv``) in a new directory
-(``js-analysis``).
+If you specify a ``scope/name`` and ``path``, then the ``path`` cannot
+be absolute. It is considered relative to the root of the CodeQL
+pack.
```
Comment (subatoi) [inline docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst:126]:
```suggestion
If you specify both a ``scope/name`` and ``path``, then the ``path`` cannot
be absolute. It is considered relative to the root of the CodeQL
pack.
```

V minor.

File: docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst [hunk 10]
```diff
@@ -95,55 +97,105 @@ You can also specify:
...
+When you specify a ``scope/name``, the ``range`` and ``path`` are
+optional. If you omit a ``range`` then the latest version of the
+specified pack is used. If you omit a ``path`` then the default query suite
+of the specified pack is used.
 
-This command runs a simple query that finds potential bugs related to unused
-variables, imports, functions, or classes---it is one of the JavaScript
-queries included in the CodeQL repository. You could run more than one query by
-specifying a space-separated list of similar paths.
+The ``path`` can be one of a ``.ql`` query file, a directory
```
Comment (subatoi) [inline docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst:118]:
```suggestion
The ``path`` can be one of: a ``.ql`` query file, a directory
```

V minor.

File: docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst [hunk 11]
```diff
@@ -95,55 +97,105 @@ You can also specify:
...
-The following examples assume your CodeQL databases have been created in a
-directory that is a sibling of your local copies of the CodeQL and CodeQL for Go
-repositories.
+Specifying which queries to run in a CodeQL pack
+------------------------------------------------
 
-Running a single query
-~~~~~~~~~~~~~~~~~~~~~~
+Query specifiers are used by ``codeql database analyze`` and other commands that operate on a set of queries.
+The complete way to specify a set of queries is in the form ``scope/name@range:path``, where:
```
Comment (subatoi) [inline docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst:107]:
```suggestion
The complete form of a query specifier is``scope/name@range:path``, where:
```

Minor, removes some superfluous language.

**Review by subatoi — COMMENTED** [review]
Hey @aeisenberg 👋 I joined Docs recently, nice to meet you! I've left some comments (mostly minor and related to style), and will be very happy to take another look after you've reviewed, or discuss anything. Thank you!

**Comment by subatoi** [comment]
@aeisenberg Felicity mentioned to me in passing that these docs follow an older style (I hadn't realised!) so if some of my comments related to that are introducing too much overhead for you in this PR, we'll be happy to take care of the edits down the road in the docs team 😀

File: docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst [hunk 12]
```diff
@@ -95,55 +97,105 @@ You can also specify:
...
+    The default query suite of the standard CodeQL query packs are ``codeql-suites/<lang>-code-scanning.qls``. Several other useful query suites can also be found in the ``codeql-suites`` directory of each pack. For example, the ``codeql/cpp-queries`` pack contains the following query suites:
 
-Use the ``--download`` flag to download the query pack if it isn't yet available locally.
+    * ``cpp-code-scanning.qls`` - Standard Code Scanning queries for C++. The default query suite for this pack.
+    * ``cpp-security-extended.qls`` - Queries from the default  ``cpp-code-scanning.qls`` suite for C++, plus lower severity and precision queries.
+    * ``cpp-security-and-quality.qls`` - Queries from ``cpp-security-extended.qls``, plus maintainability and reliability queries.
+
+    You can see the sources for these query suites in the `CodeQL repository <https://github.com/github/codeql/tree/main/cpp/ql/src/codeql-suites>`__. Query suites for other languages are similar.
+
+Examples of using ``codeql database analyze``
```
[the remark this answers is not in the document]
Comment (aeisenberg) [reply docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst:151]: How about this:

```suggestion
Examples of running database analyses
```

File: docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst [hunk 13]
```diff
@@ -15,11 +15,13 @@ For information about writing queries to run with ``database analyze``, see
 
 Before starting an analysis you must:
 
-- :doc:`Set up the CodeQL CLI <getting-started-with-the-codeql-cli>` so that it can find the queries
-  and libraries included in the CodeQL repository.
+- :doc:`Set up the CodeQL CLI <getting-started-with-the-codeql-cli>` to run commands locally.
 - :doc:`Create a CodeQL database <creating-codeql-databases>` for the source
   code you want to analyze.
 
+It is simplest to run ``database analyze`` using CodeQL packs. You can also
```
[the remark this answers is not in the document]
Comment (subatoi) [reply docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst:22]: Should this suggestion be(?):

```suggestion
The simplest way to run ``codeql database analyze`` is using CodeQL packs. You can also
```

[the remark this answers is not in the document]
Comment (aeisenberg) [reply docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst:22]: Yes!

File: docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst [hunk 14]
```diff
@@ -34,7 +36,7 @@ When you run ``database analyze``, it:
 
 You can analyze a database by running the following command::
 
-   codeql database analyze <database> --format=<format> --output=<output> <queries>
+   codeql database analyze <database> --format=<format> --output=<output> <query-specifiers>...
```
[the remark this answers is not in the document]
Comment (subatoi) [reply docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst:39]: That's fine, thanks for the context! Feel free to resolve this comment with no change.

File: docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst (hunk 13 above)
[the remark this answers is not in the document]
Comment (subatoi) [reply docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst]: Cool, applied!

File: docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst (hunk 12 above)
[the remark this answers is not in the document]
Comment (subatoi) [reply docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst:151]: LGTM

**Comment by subatoi** [comment]
> @subatoi, I'm having trouble merging your suggestions inline in the PR. It looks like there's some sort of UI but where github is complaining about multiple suggestions on the same line being committed together (which I am not doing). I'm probably going to file an issue.

@aeisenberg Let me know if you need me to commit anything if you're still having trouble!

File: docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst (hunk 13 above)
[the remark this answers is not in the document]
Comment (subatoi) [reply docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst]: This is discussed in another thread/suggestion so I'll go ahead and resolve this.

**Comment by aeisenberg** [comment]
I raised this issue: https://github.com/github/pull-requests/issues/5684

I think I can move ahead by committing suggestions one-by-one. If not, I'll just reproduce your suggestions locally. Thanks for your review!

**Comment by aeisenberg** [comment]
I think everything is resolved, but I may have missed something due to the issues I mentioned above. I squashed a bunch of the commits generated from suggestions. Please let me know if there is anything left to discuss.

File: docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst [hunk 15]
```diff
@@ -95,55 +97,105 @@ You can also specify:
...
-pack names and use the ``--download`` flag::
+pack names::
 
    codeql database analyze <database> microsoft/coding-standards@1.0.0 github/security-queries --format=sarifv2.1.0 --output=query-results.sarif --download
 
-The ``analyze`` command above runs the default suite from ``microsoft/coding-standards v1.0.0`` and the latest version of ``github/security-queries`` on the specified database.
-For further information about default suites, see ":ref:`Publishing and using CodeQL packs <publishing-and-using-codeql-packs>`".
+The ``analyze`` command above runs the default query suite of two CodeQL query packs: ``microsoft/coding-standards`` version 1.0.0 and the latest version of ``github/security-queries`` on the
+specified database. For further information about default suites, see
+":ref:`Publishing and using CodeQL packs <publishing-and-using-codeql-packs>`".
```
[the remark this answers is not in the document]
Comment (subatoi) [reply docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst]: Will resolve this as it's newer style and as discussed, this doc follows an older one.

**Review by subatoi — APPROVED** [review]
Thanks @aeisenberg! 🚀

**Pull request merged**